docs/architecture_and_project_structure.md

```diff
@@ -34,16 +34,24 @@
         - {name/common}
             - di
             - domain
-                - entity
-                - repository
-                - mappers
-            - service
-            - screens
-                - {screen_name}
-                    - widget
-                    - wm
-                    - model
-            - widgets
+                - entities
+                - repositories *(interfaces)*
+            - data
+              - repositories *(implementations)*
+              - mappers
+            - presentation
+              if feature contains multiple sub-features/screens
+                - {sub_feature_name/screen_name}
+                    - {sub_feature_name/screen_name}_widget.dart
+                    - {sub_feature_name/screen_name}_wm.dart
+                    - {sub_feature_name/screen_name}_model.dart
+              else 
+                - {sub_feature_name/screen_name}_widget.dart
+                - {sub_feature_name/screen_name}_wm.dart
+                - {sub_feature_name/screen_name}_model.dart
+
+                - widgets
+                {feature_name}_flow.dart
```

Comment:
`{feature_name}_flow.dart` is an entry point for the feature (or sub_feature) so it should be here in `else` case and inside sub-feature's folder when we have a sub-feature.